libs/ml/tests/ml/layers/multihead_attention.cpp

```diff
@@ -111,3 +113,115 @@ TYPED_TEST(MultiheadAttention, backward_test)  // Use the class as an Ops
   ASSERT_EQ(backprop_error[0].shape()[1], 20);
   ASSERT_EQ(backprop_error[0].shape()[2], 5);
 }
+
+TYPED_TEST(MultiheadAttention, saveparams_test)
+{
+  //  using DataType  = typename TypeParam::Type;
+  using LayerType = typename fetch::ml::layers::MultiheadAttention<TypeParam>;
+  using SPType    = typename LayerType::SPType;
+
+  fetch::math::SizeType n_heads   = 3;
+  fetch::math::SizeType model_dim = 6;
+
+  std::string output_name = "MultiheadAttention_Final_Transformation";
+
+  // create input data
+  TypeParam query_data = TypeParam({model_dim, 12, n_heads});
+  query_data.FillUniformRandom();
+
+  TypeParam key_data   = query_data.Copy();
+  TypeParam value_data = query_data.Copy();
+
+  // create labels data
+  TypeParam labels({6, 12, 3});
+  labels.FillUniformRandom();
+
+  // Create layer
+  LayerType layer(n_heads, model_dim);
+
+  // add label node
+  std::string label_name =
+      layer.template AddNode<fetch::ml::ops::PlaceHolder<TypeParam>>("label", {});
+
+  // Add loss function
+  std::string error_output = layer.template AddNode<fetch::ml::ops::MeanSquareErrorLoss<TypeParam>>(
+      "num_error", {output_name, label_name});
+
+  // set input and evaluate
+  layer.SetInput("MultiheadAttention_Query", query_data);
+  layer.SetInput("MultiheadAttention_Key", key_data);
+  layer.SetInput("MultiheadAttention_Value", value_data);
+
+  TypeParam prediction;
+  prediction = layer.Evaluate(output_name, true);
+
+  // extract saveparams
+  auto sp = layer.GetOpSaveableParams();
+
+  // downcast to correct type
+  auto dsp = std::dynamic_pointer_cast<SPType>(sp);
+
+  // serialize
+  fetch::serializers::MsgPackSerializer b;
+  b << *dsp;
+
+  // deserialize
+  b.seek(0);
+  auto dsp2 = std::make_shared<SPType>();
+  b >> *dsp2;
+
+  EXPECT_ANY_THROW(*(fetch::ml::utilities::BuildLayer<TypeParam, LayerType>(dsp2)));
+
+  //  // rebuild
```

Comment:
Commented out code

Reply:
It will be uncommented once shared-weight layer rebuilding works! I've added an issue for this.